common/src/View/MapDocument.cpp

```diff
@@ -1941,6 +1958,13 @@ namespace TrenchBroom {
             return executeAndStore(std::make_unique<SwapNodeContentsCommand>(commandName, std::move(nodesToSwap), std::move(linkedGroupsToUpdate)))->success();
         }
 
+        bool MapDocument::swapNodeContents(const std::string& commandName, std::vector<std::pair<Model::Node*, Model::NodeContents>> nodesToSwap) {
```

Comment:
We talked about this overload and that `swapNodeContents` needs to take the list of linked groups to update, since these linked nodes may be different depending on the operation performed. I'm a bit worried that this function might invite more use of `swapNodeContents`, and that it's too easy to not think about which linked groups need to be updated with this overload.

I would prefer to make `findLinkedGroupsToUpdate` public and remove this overload. That would force whoever wants to use `swapNodeContents` to think about what to pass for the linked groups to update, so that there is less chance to update the wrong linked groups without thinking it through.

What do you think?

Reply:
My understanding is, *not* wanting to update linked groups is a special case that only happens when transforming a linked group (or possibly other updates that deliberately desynchronize the linked groups - do you call swapNodeContents when setting protected properties?)

This suggests the public API should be `swapNodeContents(const std::string& commandName, std::vector<std::pair<Model::Node*, Model::NodeContents>> nodesToSwap, bool updateLinkedGroups);` i.e. don't make it more flexible than needed.

I'm not seeing a use case for `findLinkedGroupsToUpdate` being public, or the exsiting function that lets you specify exactly which linked groups to update.

Reply:
Also:

> I'm a bit worried that this function might invite more use of swapNodeContents

Is that a bad thing? I went through various iterations of ResizeBrushesTool while working on this PR where more of the work was done inside MapDocument. In the end, just manipulating the Brushes how I wanted (in ResizeBrushesTool) and calling `swapNodeContents` was easily the best solution, so we should be doing this more, IMO, rather than less - at least based on this experience.